### PR TITLE
CSP-81 Implements edit location option

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

File: src/SFA.DAS.Roatp.CourseManagement.Application/ProviderStandards/Commands/DeleteCourseLocations/DeleteCourseLocationsCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        public DeleteCourseLocationsCommandHandler(ILogger<DeleteCourseLocationsCommandHandler> logger, IApiClient apiClient)
+        {
+            _logger = logger;
+            _apiClient = apiClient;
+        }
+
+        public async Task<Unit> Handle(DeleteCourseLocationsCommand request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Command triggered to delete regional locations for ukprn:{ukprn} LarsCode:{larscode} from user:{userid}", request.Ukprn, request.LarsCode, request.UserId);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Application/ProviderStandards/Commands/DeleteCourseLocations/DeleteCourseLocationsCommandHandler.cs:24]: the comments in this state it's about regional locations being deleted, but it does read from the command name etc to be all courselocations (provider, regional, national).   If it's all please amend comments, if it's just regions, leave as is.

File: src/SFA.DAS.Roatp.CourseManagement.Application.UnitTests/ProviderStandards/Queries/GetStandardDetails/GetStandardDetailsQueryResultTests/LocationOptionTests.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        [Test]
+        public void AnyNonProviderLocations_ReturnsEmployer()
+        {
+            _sut.ProviderCourseLocations.Remove(_providerLocation);
+
+            _sut.LocationOption.Should().Be(LocationOption.EmployerLocation);
+        }
+
+        [Test]
+        public void HasProviderAndNationalLocations_ReturnsEmployer()
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Application.UnitTests/ProviderStandards/Queries/GetStandardDetails/GetStandardDetailsQueryResultTests/LocationOptionTests.cs:55]: rename test to be about Both

File: src/SFA.DAS.Roatp.CourseManagement.Web/AppStart/AddServiceRegistraionsExtension.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.Extensions.Configuration;
```
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web/AppStart/AddServiceRegistraionsExtension.cs:1]: Fixed

File: src/SFA.DAS.Roatp.CourseManagement.Application/ProviderStandards/Commands/DeleteCourseLocations/DeleteCourseLocationsCommandHandler.cs (hunk 1 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Application/ProviderStandards/Commands/DeleteCourseLocations/DeleteCourseLocationsCommandHandler.cs:24]: Updated log

File: src/SFA.DAS.Roatp.CourseManagement.Application.UnitTests/ProviderStandards/Queries/GetStandardDetails/GetStandardDetailsQueryResultTests/LocationOptionTests.cs (hunk 2 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Application.UnitTests/ProviderStandards/Queries/GetStandardDetails/GetStandardDetailsQueryResultTests/LocationOptionTests.cs:55]: Done

File: src/SFA.DAS.Roatp.CourseManagement.Application.UnitTests/ProviderStandards/Queries/GetStandardDetails/GetStandardDetailsQueryResultTests/LocationOptionTests.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            _sut.ProviderCourseLocations.Remove(_regionalLocation);
+
+            _sut.LocationOption.Should().Be(LocationOption.Both);
+        }
+
+        [Test]
+        public void HasProviderAndRegionalLocations_ReturnsEmployer()
```
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Application.UnitTests/ProviderStandards/Queries/GetStandardDetails/GetStandardDetailsQueryResultTests/LocationOptionTests.cs:63]: duplicate comment.

File: src/SFA.DAS.Roatp.CourseManagement.Web/Validators/EditLocationOptionViewModelValidator.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+﻿using FluentValidation;
+using SFA.DAS.Roatp.CourseManagement.Domain.Models;
+using SFA.DAS.Roatp.CourseManagement.Web.Models;
+
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Validators/EditLocationOptionViewModelValidator.cs:4]: add text as constant, then use that reference in the appropriate test

**Review by MarkFCain — DISMISSED** [review]

File: src/SFA.DAS.Roatp.CourseManagement.Web/Validators/EditLocationOptionViewModelValidator.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+namespace SFA.DAS.Roatp.CourseManagement.Web.Validators
+{
+    public class EditLocationOptionViewModelValidator : AbstractValidator<EditLocationOptionViewModel>
+    {
+        public EditLocationOptionViewModelValidator()
+        {
+            RuleFor(m => m.LocationOption)
+                .NotEqual(LocationOption.None)
+                .WithMessage("Select where you will deliver this standard");
+        }
```
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web/Validators/EditLocationOptionViewModelValidator.cs:16]: Done

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**